api/src/services/occurrence-service.ts

```diff
@@ -236,4 +71,42 @@ export class OccurrenceService extends DBService {
   async updateDWCSourceForOccurrenceSubmission(submissionId: number, jsonData: string): Promise<number> {
     return await this.occurrenceRepository.updateDWCSourceForOccurrenceSubmission(submissionId, jsonData);
   }
+
+  /**
+   * Query builder to find spatial component by given criteria
+   *
+   * @param {ISpatialComponentsSearchCriteria} criteria
+   * @return {*}  {Promise<ISubmissionSpatialComponent[]>}
+   * @memberof SpatialService
+   */
+  async findSpatialMetadataBySubmissionSpatialComponentIds(
+    submissionSpatialComponentIds: number[]
+  ): Promise<GeoJsonProperties[]> {
+    const response = this.occurrenceRepository.findSpatialMetadataBySubmissionSpatialComponentIds(
+      submissionSpatialComponentIds
+    );
+
+    return (await response).map((row) => row.spatial_component_properties);
```

Comment:
Can we move this `await` to the line above: `const response = await this.occurrenceRepository....`